### PR TITLE
Redesign Page Header

### LINKED ISSUE #6205: Redesign: Page header
## Feature Description

Update page header based on [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=1009%3A4323).

Pay attention to both sizing and spacing including that of the entire header.

- [ ] Update page header sizing and spacing
- [ ] Update primary button design to use design system primary button
- [ ] Update secondary buttons to be icons (using design system icon button)

| Current design | New design |
|-|-|
| Screen Shot 2021-02-04 at 22 44 02 | Screen Shot 2021-02-04 at 22 43 54 |

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Summary
Updates the action buttons in the page header.

## To-do

- [x] Missing icon for "Switch to Draft"

## User-facing changes
Before:
Screenshot 2021-02-09 at 14 14 44

After:
Screenshot 2021-03-05 at 21 00 09

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Open a new story
2. Verify the page header buttons look as in [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories-Post-Stable?node-id=6515%3A521262)
3. Publish the story and verify there is "Save to Draft" button now instead of "Save draft" button.
4. Verify that during saving, a spinner is displayed around the save draft/switch to draft button.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6205

### CODE REVIEW

File: assets/src/design-system/components/button/index.js
Comment (miina) [inline assets/src/design-system/components/button/index.js]: Changed this here since apparently there is `border: 2px solid transparent` applied additionally in a separate place which increases the width and height of the button by `4px` making it larger than it's supposed to be. Perhaps this should be addressed in the place where the border is assigned instead:
https://github.com/google/web-stories-wp/blob/446bbd5f30075f8f74884a1f77894749a49b746c/assets/src/design-system/theme/helpers/outline.js#L48

File: assets/src/edit-story/components/header/buttons/switchToDraft.js
Comment (miina) [inline assets/src/edit-story/components/header/buttons/switchToDraft.js]: Icon TBD

File: assets/src/design-system/components/button/index.js
Comment (barklund) [reply assets/src/design-system/components/button/index.js]: Yeah, that sounds like something that should be detracted only when and if the outline is applied.

Comment (swissspidy) [reply assets/src/design-system/components/button/index.js]: I think I saw somewhere in another PR that this `border: 2px solid transparent;` might/should get removed. Worth synching with pea pod on it.

Comment (miina) [reply assets/src/design-system/components/button/index.js]: @BrittanyIRL -- do you have any insights regarding this border and its potential removal?

Comment (BrittanyIRL) [reply assets/src/design-system/components/button/index.js]: Pascal has a great memory, he's right that `border: 2px solid transparent;` in helpers/outline needs to be removed. That got readded in a rebase and snuck through.  here's a small PR to get that style out of here: https://github.com/google/web-stories-wp/pull/6303

File: assets/src/edit-story/components/header/buttons/index.js
Comment (swissspidy) [inline assets/src/edit-story/components/header/buttons/index.js:52]: Why is this needed / what is this now positioned to in relation?

Comment (miina) [reply assets/src/edit-story/components/header/buttons/index.js:52]: This will now be positioned on top of the first icon of the buttons area, either the "Save Draft" or "Switch to Draft":
Screenshot 2021-02-11 at 13 47 51

We could add a wrapper around the first icon and the spinner together to make it more explicit. WDYT?

Comment (barklund) [reply assets/src/edit-story/components/header/buttons/index.js:52]: Yeah, put them together in a wrapper more explicitly - that would make the intent more obvious.

File: assets/src/edit-story/components/header/buttons/preview.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/preview.js]:
```suggestion
        hasTail
```

I actually thought we had an eslint rule for this?

@swissspidy, could we add [`react/jsx-boolean-value`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md) (config'ed to `never`) if we haven't already?

File: assets/src/design-system/icons/action/index.js
Comment (barklund) [inline assets/src/design-system/icons/action/index.js]: You really ought to merge in the icon PR from #6311, as this icon is already imported, but in the proper size and bounding box. That'll also ensure that the icons scale properly relative to each other.

File: assets/src/edit-story/components/header/buttons/index.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/index.js:65]: I actually don't see this spinner for the _Save draft_ button? I see it for the the other buttons correctly (_Publish_, _Update_, _Switch to draft_), but not for _Save draft_. Does that make sense?

File: assets/src/edit-story/components/header/buttons/switchToDraft.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/switchToDraft.js]: I'm afraid that someone might have changed the designs while you're working on them.

[Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=327%3A10229):
Screen Shot 2021-02-12 at 00 08 08

Checking with @samitron7 [here](https://xwp.slack.com/archives/C01ESBB8ZN1/p1613106676206900).

File: assets/src/edit-story/components/header/buttons/update.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/update.js]: This one has to be updated. If the user is saving or uploading, the button _**has**_ to be disabled. You shouldn't be able to save under those circumstances regardless of the presence of meta-boxes. It's only `hasNewChanges` that can be overridden by `hasMetaboxes`:

```js
              isSaving || isUploading || (!hasNewChanges && !hasMetaBoxes)
```

But it might make sense to move this to a (positively named) variable, pull it up top, and affix a wordy comment about what's in here and why. I hate getting confused by these - especially negative flags 😂 😂 .

So, something like:

```js
  // The button is enabled only if we're not already saving nor uploading. And
  // then only if there are new changes or the story has meta-boxes – as they
  // can update without us knowing it.
  const isEnabled = !isSaving && !isUploading && (hasNewChanges || hasMetaBoxes)
  ...
  disabled={!isEnabled}
```

File: assets/src/edit-story/components/header/buttons/switchToDraft.js
Comment (barklund) [reply assets/src/edit-story/components/header/buttons/switchToDraft.js]: Turns out the design is not changing. Lucky you 😁  🍾 .

**Review by barklund — COMMENTED** [review]
This looks great! But we need the new icons imported here - they're better balanced and correctly sized.

But it looks great - and those new tooltips are smooth and fine!

File: assets/src/edit-story/components/header/buttons/switchToDraft.js
Comment (barklund) [reply assets/src/edit-story/components/header/buttons/switchToDraft.js]: And then again maybe it is, but stick to the current path for now.

(see above thread for more)

File: assets/src/design-system/icons/action/index.js
Comment (miina) [reply assets/src/design-system/icons/action/index.js]: Will do, I'm hoping it will get merged soon so I can keep the changed files of this PR under 100 😛 

Waiting for the design confirmation on the icons-vs-text for buttons, too.

File: assets/src/edit-story/components/header/buttons/preview.js
Comment (swissspidy) [reply assets/src/edit-story/components/header/buttons/preview.js]: Sure. That rule is not in the recommended ruleset, so needs to be manually added.

File: assets/src/edit-story/components/header/buttons/index.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/index.js]: This space varies between 8 and 12px wide depending on browser size. The cleanest way is probably to use a CSS prop set on the wrapper based on window width or workspace width. Workspace width is available from some provider I think.

File: assets/src/edit-story/components/header/buttons/index.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/index.js]:
```js
            {isDraft ? <Update /> : <SwitchToDraft />}
```

File: assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js]: Do we still want an alert icon here? The [design](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2172%3A244419) doesn't have one anymore... And if we do, should we use the circular one rather than the triangle one - given that the triangle is used for warnings and the circle for errors.

Also, should the tooltip be laid out as in the design?
Screen Shot 2021-02-15 at 16 13 59

Finally, the designs mention that the user can click on the icon to go to the checklist, but not sure how that's going to work. Maybe we just want the tooltip to come with a clickable link, but that might also be tricky?

Comment (barklund) [reply assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js]: Also, using the boolean'ess of `tooltip` to toggle the icon seems confusing. Could we add a boolean variable instead?

**Review by barklund — COMMENTED** [review]
Looks good, just a few adjustments. Especially around the warning indicator - the designs are a bit ambiguous but also a bit ambitious with clickable links etc.

File: assets/src/edit-story/components/header/buttons/update.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/update.js:88]: Should we add the shortcut `mod+s` here as well?

File: assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js
Comment (miina) [reply assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js]: > Do we still want an alert icon here?

IDK, it existed before but perhaps not anymore 🤷  I think this can be done separately though once it has been confirmed -- the whole pre-publishing adjustment part and the tooltip with link, we can perhaps for now keep that part as it was, WDYT?

File: assets/src/edit-story/components/header/buttons/index.js
Comment (miina) [reply assets/src/edit-story/components/header/buttons/index.js]: Hm, I can actually see only 12px currently, could you share a link where the 8px can be seen? (Although this might become irrelevant since the design is not fully confirmed yet.)

File: assets/src/edit-story/components/header/buttons/update.js
Comment (miina) [reply assets/src/edit-story/components/header/buttons/update.js:88]: `useGlobalKeyDownEffect` already exists in this file for `mod+s`, see a bit above (unchanged part of the file). If that's what you meant.

Comment (barklund) [reply assets/src/edit-story/components/header/buttons/update.js:88]: I meant to show the actual shortcut in the tooltip itself by adding `shortcut="mod+s"` here.

File: assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js
Comment (barklund) [reply assets/src/edit-story/components/header/buttons/buttonWithChecklistWarning.js]: Create a P2 follow-up ticket to sort that stuff out. Makes sense to delay until layer, as it's far from trivial and probably not that important.

File: assets/src/edit-story/components/header/buttons/index.js
Comment (barklund) [reply assets/src/edit-story/components/header/buttons/index.js]: It happens on the smaller breakpoints:
Screen Shot 2021-02-17 at 01 41 03

Comment (barklund) [reply assets/src/edit-story/components/header/buttons/index.js]: (and the [link](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=327%3A10229) you asked for)

File: assets/src/edit-story/components/header/buttons/update.js
Comment (barklund) [inline assets/src/edit-story/components/header/buttons/update.js]: So, no more checklist warning icon? I assume that's per agreement, I just haven't seen that convo.

**Review by barklund — APPROVED** [review]
Looks awesome, good to go! And this is one of the most visible updates of the redesign, so GO GO GO 🎉 🚢 🚀

File: assets/src/edit-story/components/header/buttons/update.js
Comment (barklund) [reply assets/src/edit-story/components/header/buttons/update.js]: Oh, is it [this convo](https://github.com/google/web-stories-wp/pull/6277#discussion_r576425810)? I think we need UX approval to remove the icon - just to be sure!

Comment (miina) [reply assets/src/edit-story/components/header/buttons/update.js]: [Confirming](https://xwp.slack.com/archives/C01ESBB8ZN1/p1614930022067400) with UX.

**Comment by swissspidy** [comment]
Not 100% sure but seems like e2e tests might need updating now with the updated page header. Just re-ran them, so let's see.

**Pull request merged**
